### ISSUE TITLE
ACLs and shares were lost after make a files:scan --all

### ISSUE THREAD

**Issue by cyrzero** [nextcloud/groupfolders#2931]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Make some ACLs and shares on folders/files in a groupfolder.
2. Do an occ files:scan --all
4. See all shares and rights in the groupfolder

### Expected behaviour
occ files:scan --all scan all the files of all the users and the groups folders. 

The ACLs and shares should steal in place in the folders and subfolders.

### Actual behaviour
After the files:scan all the ACLs and shares who's been created by the users disappeared. 
All the Nextcloud Clients try to synchronize the entire folder tree, then load the server. 

We need to recreate all the rights in the folders and shares were lost.

### Server configuration

**Operating system:**
Debian GNU/Linux 12

**Web server:**
Apache 2

**Database:**
10.11.6-MariaDB-0+deb12u1

**PHP version:**
Php 8.2

**Nextcloud version:** (see Nextcloud admin page)
[Nextcloud Hub 6] (27.1.8)

**Group folders version:**
15.3.7
**Updated from an older Nextcloud/ownCloud or fresh install:**
Updated from previous versions.
**Where did you install Nextcloud from:**
Release and updater. 
**Are you using external storage, if yes which one:** local/s3/smb/sftp/...
No
**Are you using encryption:** yes/no
No
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/Saml/...
LDAP/ActiveDirectory
### Client configuration
**Browser:**
Firefox
**Operating system:**
Linux Windows.
### Logs
There are no obvious logs during the scan files, who take 5 hours because each user will rescan the groupfolder. 

#### Web server error log
<details>
<summary>Web server error log</summary>

No errors log.
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

</details>

There are no logs in particular who speak of delete ACLs or shares... 

#### Browser log
<details>
<summary>Browser log</summary>

The client browser is not important. 

</details>

**Comment by Ramalama2** [comment]
i thought of this problem while files:scan took a lot longer to scan on a user without personal files.
Searched for issues while scanning and im hoping now that those 2 users didn't lost their permissions.

Hopefully this can be fixed, its pretty dangerous otherwise!

**Comment by joshtrichards** [comment]
This definitely sounds bad, but I'm unable to reproduce this behavior so far. 

@Ramalama2 It's unclear from your comment... did you experience this too or you were merely concerned about the possibility?

@Panzhunter & @ynott - I see you both gave this report a thumbs up. Are you experiencing the same behavior?

**Comment by Ramalama2** [comment]
> This definitely sounds bad, but I'm unable to reproduce this behavior so far.
> 
> @Ramalama2 It's unclear from your comment... did you experience this too or you were merely concerned about the possibility?
> 
> @Panzhunter & @ynott - I see you both gave this report a thumbs up. Are you experiencing the same behavior?

No, the files scan all, didnt broke anything. But i never ran it and aborted directly after i readed about this issue.
My problem was that the scan all command took over 20 hours and stuck on the first user that had access to a shared groupfolder.
Then i searched about any issues related to this, found this and stopped immediately files scan.
And never ran it again.

Cheers

**Comment by MauritzFunke** [comment]
I've got the same issue. After a server issue I rescanned all files because the database cache and the real files were inconsistent. After scanning all files all shares disapeared.

**Comment by provokateurin** [comment]
@MauritzFunke did you manually modify oc_filecache at some point?

**Comment by MauritzFunke** [comment]
Yesn't, I restored a db backup after data corruption due to some errors while migrating VM hosts, we had some data loss (a few cache files that were synced via the desktop app (thats why I rescanned all files)

**Comment by MauritzFunke** [comment]
This was the output of the scan command:
```
+---------+--------+-----+----------+---------+--------+--------------+
| Folders | Files  | New | Updated  | Removed | Errors | Elapsed time |
+---------+--------+-----+----------+---------+--------+--------------+
| 90576   | 611634 | 0   | 17852091 | 43      | 71     | 00:23:07     |
+---------+--------+-----+----------+---------+--------+--------------+
```

**Comment by JohnWalkerx** [comment]
> Yesn't, I restored a db backup after data corruption due to some errors while migrating VM hosts, we had some data loss (a few cache files that were synced via the desktop app (thats why I rescanned all files)

Same happend to me yesterday. I migrated to another server. Copied files, restored database and no files in the group folders where shown.
Rescanned, files got shown again but Advanced ACL where lost.
Also shared links lost and can't create link shares anymore.
That is a disaster!

Anyone an idea what could cause this?
